### PR TITLE
Backport master changes to next

### PR DESCRIPTION
**Pull request by slax57**
We will do this regularly to make the conflicts resolution easier once the v5 is released.

### CODE REVIEW

File: docs/InfiniteList.md [hunk 1]
```diff
@@ -161,4 +161,111 @@ export const BookList = () => (
...
+The previous example leverages [`<WithListContext>`](./WithListContext.md) to grab the data that `<ListBase>` stores in the `ListContext`.
+
+If you don't need the `ListContext`, you can use the `useInfiniteListController` hook, which does the same data fetching as `<InfiniteListBase>` but lets you render the content.
+
+```jsx
+import { useInfiniteListController } from 'react-admin';
+import { Card, CardContent, Container, Stack, Typography } from '@mui/material';
+
+const ProductList = () => {
+    const { isLoading, data } = useInfiniteListController();
```
Comment (fzaninotto) [inline docs/InfiniteList.md:251]:
```suggestion
    const { isPending, data } = useInfiniteListController();
```

File: docs/InfiniteList.md [hunk 2]
```diff
@@ -161,4 +161,111 @@ export const BookList = () => (
...
+    return (
+        <Container>
+            <Typography variant="h4">All products</Typography>
+                {!isLoading && (
```
Comment (fzaninotto) [inline docs/InfiniteList.md:255]:
```suggestion
                {!isPending && (
```

File: docs/InfiniteList.md [hunk 3]
```diff
@@ -161,4 +161,111 @@ export const BookList = () => (
...
+```jsx
+import { InfiniteListBase, InfinitePagination, WithListContext } from 'react-admin';
+import { Card, CardContent, Container, Stack, Typography } from '@mui/material';
+
+const ProductList = () => (
+    <InfiniteListBase>
+        <Container>
+            <Typography variant="h4">All products</Typography>
+            <WithListContext render={({ isLoading, data }) => (
+                    !isLoading && (
```
Comment (fzaninotto) [inline docs/InfiniteList.md:224]:
```suggestion
            <WithListContext render={({ isPending, data }) => (
                    !isPending && (
```

File: docs/Features.md [hunk 4]
```diff
@@ -271,6 +271,98 @@ We have made many improvements to this default layout based on user feedback. In
...
+          size="small"
+          dataSource={data}
+          columns={columns}
+          pagination={{ current: page, pageSize: 10, total }}
+          onChange={handleTableChange}
+        />
+      </Card>
+    </>
+  );
+};
```
Comment (fzaninotto) [inline docs/Features.md:323]:
```suggestion
const PostList = () => {
  const { data, page, total, setPage, isPending } = useListController({
    sort: { field: 'published_at', order: 'DESC' },
    perPage: 10,
  });
  const handleTableChange = (pagination) => {
    setPage(pagination.current);
  };
  return (
    <>
      <div style={{ margin: 10, textAlign: 'right' }}>
        <Link to="/posts/create">
          <Button icon={<PlusOutlined />}>Create</Button>
        </Link>
      </div>
      <Card bodyStyle={{ padding: '0' }} loading={isPending}>
        <Table
          size="small"
          dataSource={data}
          columns={columns}
          pagination={{ current: page, pageSize: 10, total }}
          onChange={handleTableChange}
        />
      </Card>
    </>
  );
};
```

File: docs/List.md [hunk 5]
```diff
@@ -1081,3 +1081,158 @@ const ProductList = () => (
...
+                            ))}
+                        </Stack>
+                    )
+                )} />
+            <WithListContext render={({ isLoading, total }) => (
+                !isLoading && <Typography>{total} results</Typography>
+            )} />
+        </Container>
+    </ListBase>
+);
```
Comment (fzaninotto) [inline docs/List.md:1205]:
```suggestion
const ProductList = () => (
    <ListBase>
        <Container>
            <Typography variant="h4">All products</Typography>
            <WithListContext render={({ isPending, data }) => (
                    !isPending && (
                        <Stack spacing={1}>
                            {data.map(product => (
                                <Card key={product.id}>
                                    <CardContent>
                                        <Typography>{product.name}</Typography>
                                    </CardContent>
                                </Card>
                            ))}
                        </Stack>
                    )
                )} />
            <WithListContext render={({ isPending, total }) => (
                !isPending && <Typography>{total} results</Typography>
            )} />
        </Container>
    </ListBase>
);
```

File: docs/List.md [hunk 6]
```diff
@@ -1081,3 +1081,158 @@ const ProductList = () => (
...
+                                    <Typography>{product.name}</Typography>
+                                </CardContent>
+                            </Card>
+                        ))}
+                    </Stack>
+                )}
+            {!isLoading && <Typography>{total} results</Typography>}
+        </Container>
+    );
+};
```
Comment (fzaninotto) [inline docs/List.md:1235]:
```suggestion
const ProductList = () => {
    const { isPending, data, total } = useListController();
    return (
        <Container>
            <Typography variant="h4">All products</Typography>
                {!isPending && (
                    <Stack spacing={1}>
                        {data.map(product => (
                            <Card key={product.id}>
                                <CardContent>
                                    <Typography>{product.name}</Typography>
                                </CardContent>
                            </Card>
                        ))}
                    </Stack>
                )}
            {!isPending && <Typography>{total} results</Typography>}
        </Container>
    );
};
```

File: docs/useListController.md [hunk 7]
```diff
@@ -5,13 +5,89 @@ title: "useListController"
...
+    setPage(pagination.current);
+  };
+  return (
+    <>
+      <div style={{ margin: 10, textAlign: 'right' }}>
+        <Link to="/posts/create">
+          <Button icon={<PlusOutlined />}>Create</Button>
+        </Link>
+      </div>
+      <Card bodyStyle={{ padding: '0' }} loading={isLoading}>
```
Comment (fzaninotto) [inline docs/useListController.md:48]:
```suggestion
  const { data, page, total, setPage, isPending } = useListController({
    sort: { field: 'published_at', order: 'DESC' },
    perPage: 10,
  });
  const handleTableChange = (pagination) => {
    setPage(pagination.current);
  };
  return (
    <>
      <div style={{ margin: 10, textAlign: 'right' }}>
        <Link to="/posts/create">
          <Button icon={<PlusOutlined />}>Create</Button>
        </Link>
      </div>
      <Card bodyStyle={{ padding: '0' }} loading={isPending}>
```

**Review by fzaninotto — CHANGES_REQUESTED** [review]

File: docs/useListController.md (hunk 7 above)
Comment (slax57) [reply docs/useListController.md:48]: nice catches!

**Comment by slax57** [comment]
We'll wait for #9481 to be merged before merging this one

**Comment by slax57** [comment]
Back to RFR

**Review by fzaninotto — APPROVED** [review]

**Pull request merged**